domain/infosync/info.go

```diff
@@ -82,17 +83,30 @@ type ServerVersionInfo struct {
 	GitHash string `json:"git_hash"`
 }
 
-var globalInfoSyncer *InfoSyncer
+var globalInfoSyncer atomic.Value
+
+func getGlobalInfoSyncer() (*InfoSyncer, error) {
+	v := globalInfoSyncer.Load()
```

Comment:
Every domain overwrites `InfoSyncer` each other, and then the last written `InfoSyncer` takes affect in all domains. Does it matter in test cases?

Reply:
Can you put the `globalInfoSyncer` into `domain`, and so each domain has one copy of `InfoSyncer`?

Reply:
domain already has the `infosyncer`.
Ye, you are right, maybe I should use the `infoSyncer` in the domain.

Reply:
Oh, I got it, I used to try to do this like you said, but it will has `circle import` problem between `domain` and `infoschema`.

Reply:
I try to put `infosyncer` in session, too many place need changed to get the `session`....

Reply:
Please add some comments here to explain why.

Reply:
done.